internal/persistence/persistors/mysqlstore/store_organization_test.go

```diff
@@ -0,0 +1,497 @@
+				require.NotNil(t, paginated.Organizations, "unexpected nil organization")
+				require.NotNil(t, paginated.Pagination, "unexpected nil pagination")
+				assert.True(t, len(paginated.Organizations) == 0, "unexpected greater than 1 organization")
+				assert.True(t, paginated.Pagination.RowCount == 0, "unexpected count to be greater than 1 organization")
+
+				modelhelpers.AssertNonEmptyOrganizations(t, paginated.Organizations)
+			},
+		},
+	}
+}
+
+func Test_GetOrganizations(t *testing.T) {
+	for _, testCase := range getTestCasesGetOrganizations() {
+		t.Run(testCase.name, func(t *testing.T) {
+			db, cp, cleanup := mysqlhelper.TestGetMockMariaDB(t)
+			defer cleanup()
+			require.NotNil(t, testCase.mutations, "unexpected nil mutations")
+			require.NotNil(t, testCase.assertions, "unexpected nil assertions")
+
+			cfg := &Config{
+				Logger:        testLogger,
+				QueryTimeouts: testQueryTimeouts,
+			}
+
+			m, err := New(cfg)
+			require.NoError(t, err, "unexpected error")
+			require.NotNil(t, m, "unexpected nil")
+
+			txHandler, err := mysqltx.New(&mysqltx.Config{
+				Logger:       testLogger,
+				Db:           db,
+				DatabaseName: cp.Database,
+			})
+			require.NoError(t, err, "unexpected error creating the tx handler")
+
+			txHandlerDb, err := txHandler.Db(testCtx)
+			require.NoError(t, err, "unexpected error fetching the db from the tx handler")
+			require.NotNil(t, txHandlerDb, "unexpected nil tx handler db")
+
+			testCase.mutations(t, db)
+			paginated, err := m.GetOrganizations(testCtx, txHandlerDb, testCase.filter)
+			testCase.assertions(t, db, paginated, err)
+		})
+	}
+}
+
+type deleteOrganizationTestCase struct {
+	name       string
+	id         int
+	assertions func(t *testing.T, db *sqlx.DB, id int, err error)
+	mutations  func(t *testing.T, db *sqlx.DB)
+}
+
+func getDeleteOrganizationTestCases() []deleteOrganizationTestCase {
+	return []deleteOrganizationTestCase{
+		{
+			name: "success",
+			id:   1,
+			assertions: func(t *testing.T, db *sqlx.DB, id int, err error) {
+				require.Nil(t, err, "unexpected non-nil error")
+				entry, err := mysqlmodel.FindOrganization(context.TODO(), db, id)
+				require.NoError(t, err, "unexpected error fetching the organization")
+
+				assert.Equal(t, false, entry.IsActive)
+			},
+			mutations: func(t *testing.T, db *sqlx.DB) {
+				entry := mysqlmodel.Organization{
+					Name:     "test organization",
+					IsActive: true,
+				}
+				err := entry.Insert(context.TODO(), db, boil.Infer())
+				assert.NoError(t, err, "unexpected insert error")
+			},
+		},
+	}
+}
+
+func Test_DeleteOrganization(t *testing.T) {
+	for _, testCase := range getDeleteOrganizationTestCases() {
+		db, cp, cleanup := mysqlhelper.TestGetMockMariaDB(t)
+		t.Run(testCase.name, func(t *testing.T) {
+			require.NotNil(t, testCase.mutations, "unexpected nil mutations")
+			require.NotNil(t, testCase.assertions, "unexpected nil assertions")
+
+			defer cleanup()
```

Comment:
move db, cp, cleanup := mysqlhelper.TestGetMockMariaDB(t) inside the function, and always run defer first before any require, so it'll look like:

t.Run(testCase.name, func(t *testing.T) {
// Now is inside the function
db, cp, cleanup := mysqlhelper.TestGetMockMariaDB(t)
                        // called defer immediately
			defer cleanup()
			require.NotNil(t, testCase.mutations, "unexpected nil mutations")
			require.NotNil(t, testCase.assertions, "unexpected nil assertions")

Reply:
Copy that. Thank you for the hint

Reply:
all right! Ensure also that this pattern is consistent across all tests